packages/libs/deer/src/lib.rs

```diff
@@ -6,4 +6,212 @@
 #![allow(clippy::module_name_repetitions)]
 #![forbid(unsafe_code)]
 
+use alloc::string::String;
+
+use error_stack::{IntoReport, Result};
+use num_traits::ToPrimitive;
+
+pub use crate::{error::Error, number::Number};
+
+mod error;
+mod number;
+
 extern crate alloc;
+
+/// Lookup of Type
+///
+/// This is used as the signal for the type that is currently present, [`Deserialize`]
+/// implementations may use this to special case certain values.
+#[derive(Debug, Copy, Clone)]
+pub enum PrimitiveType {
```

Comment:
Defining the type by a trait seems more powerful (see `serde::de::Visitor::Value`) and is less specific. Any specific reason you chose a concrete type here?

Reply:
removed in [c8a8e45](https://github.com/hashintel/hash/pull/1170/commits/c8a8e452b4d72c95b18b136f85172f35768d77b8)